pkgs/native_assets_builder/lib/src/locking/locking.dart

```diff
@@ -17,8 +19,10 @@ Future<T> runUnderDirectoriesLock<T>(
     return await callback();
   }
   return await runUnderDirectoryLock(
+    fileSystem,
     directories.first,
```

Comment:
Ditto, the directories already should have a file system. Either take the fileSystem from there or add asserts that the file system is equal.

Reply:
Ditto. I think both of those options are somewhat ugly:

* Yes, the `File`/`Directory` *happen to expose* the `FileSystem` but conceptually they maybe shouldn't, as that creates circularity between those classes. The `FileSystem` is a higher-level abstraction than `File`.
* It creates ambiguity: If we pass a `List<Directory>` we implicitly assume they all belong to the same `FileSystem`, this implicit assumption - whether asserted or not - is somewhat ugly.

So a cleaner way is to pass `FileSystem` and `Uri`s around. Then the implementation can ensure it operates on the same file system on all the `Uri`s it's given.

Hope that's ok with you.

Reply:
> So a cleaner way is to pass `FileSystem` and `Uri`s around. Then the implementation can ensure it operates on the same file system on all the `Uri`s it's given.

SGTM